helm/hades/readme.md

```diff
@@ -0,0 +1,124 @@
+# Hades Helm Chart
+
+Deploy the **Hades** build system (API, Scheduler, and NATS broker) into any Kubernetes cluster using Helm.
+
+---
+
+## Contents
+
+* [Overview](#overview)
+* [Prerequisites](#prerequisites)
+* [Quick Start](#quick-start)
+* [Configuration](#configuration)
+
+    * [Values Reference](#values-reference)
+* [Upgrade & Rollback](#upgrade--rollback)
+* [Uninstall](#uninstall)
+* [Development](#development)
+
+---
+
+## Overview
+
+This chart bundles the core services of Hades:
+
+| Component           | Description                                                                      |
+|---------------------|----------------------------------------------------------------------------------|
+| **hades-api**       | Processes and validates the request and produce the build request as NATS events |
+| **hades-scheduler** | Consumes NATS events and spawns per-exercise runner Pods                         |
+| **hades-nats**      | Embedded [NATS JetStream](https://nats.io) message broker (sub-chart)            |
+
+The Scheduler operates in **`serviceaccount` mode** by default, using Kubernetes-native in-cluster credentials to authenticate and create jobs.
+
+> ⚠️ The previously supported `kubeconfig` mode is not implemented in this version.
+
+---
+
+## Prerequisites
+
+* Kubernetes **v1.25+**
+* Helm **v3.12+**
+
+---
+
+## Quick Start
+
+```bash
+# 1. Create target namespace if the namespace does not exist
+kubectl create namespace hades
+
+# 2. Install the chart using serviceaccount mode (default)
+helm upgrade --install hades ./helm/hades -n hades --create-namespace
+
```

Comment:
When following this tutorial (with the previous things fixed) i get the following error: 
```
2025/07/22 19:52:24 INFO Creating namespace in Kubernetes
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc8 pc=0x113aae4]

goroutine 1 [running]:
k8s.io/client-go/kubernetes.(*Clientset).CoreV1(...)
	/go/pkg/mod/k8s.io/client-go@v0.34.0-alpha.2/kubernetes/clientset.go:323
github.com/ls1intum/hades/hadesScheduler/k8s.createNamespace({0x168e580, 0x1f61520}, 0x0, {0xc000046026, 0xa})
	/app/HadesScheduler/k8s/namespace.go:17 +0xe4
github.com/ls1intum/hades/hadesScheduler/k8s.NewK8sScheduler()
	/app/HadesScheduler/k8s/k8s.go:55 +0x1cf
main.main()
	/app/HadesScheduler/main.go:64 +0x3ca

```

Reply:
👍